src/libraries/System.IO.Packaging/src/System/IO/Packaging/OrderedDictionary.cs

```diff
@@ -0,0 +1,75 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections;
+using System.Collections.Generic;
+
+namespace System.IO.Packaging
+{
+    /// <summary>
+    /// A collection that ensures uniqueness among a list of elements while maintaining the order in which the elements were added.
+    /// This is similar to <see cref="OrderedDictionary{TKey, TValue}"/>, but the items will not be sorted by a comparer but rather retain the
+    /// order in which they were added while still retaining good lookup, insertion, and removal.
+    /// </summary>
+    internal class OrderedDictionary<TKey, TValue> : IEnumerable<TValue>
+    {
+        private readonly Dictionary<TKey, LinkedListNode<TValue>> _dictionary;
+        private readonly LinkedList<TValue> _order;
+
+        public OrderedDictionary(int initialCapacity, IEqualityComparer<TKey> comparer)
+        {
+            _dictionary = new Dictionary<TKey, LinkedListNode<TValue>>(initialCapacity, comparer);
+            _order = new LinkedList<TValue>();
+        }
+
+        public bool Contains(TKey key) => _dictionary.ContainsKey(key);
+
+        public bool Add(TKey key, TValue value)
+        {
+            if (_dictionary.ContainsKey(key))
+            {
+                return false;
+            }
+
+            _dictionary.Add(key, _order.AddLast(value));
+            return true;
+        }
+
+        public void Clear()
+        {
+            _dictionary.Clear();
+            _order.Clear();
+        }
+
+        public bool Remove(TKey key)
+        {
+            if (_dictionary.TryGetValue(key, out var value))
```

Comment:
Can this use the Remove overload that gives back the associated value as an out?

Reply:
This has to compile against .NET Standard 1.3 and 2.0 and that's not available. Should I do an #if/#def?

Reply:
Probably not worth it just for this.